toucan_connectors/trello/trello_connector.py

```diff
@@ -0,0 +1,148 @@
+"""
+Provide a trello connector.
+
+Check https://developers.trello.com/docs/get-started for the official API documentation.
+
+The connector allow you to list in a dataset all the cards of a board with a chosen set of "fields".
+The available fields are listed in the `Field ` object.
+
+The dataset has one line by card and one columns by fields
+"""
+
+import pandas as pd
+from typing import List
+from enum import Enum
+import requests
+import warnings
+
+from toucan_connectors.toucan_connector import ToucanConnector, ToucanDataSource
+
+
+class Fields(str, Enum):
+    name = 'name'
+    url = 'url'
+    lists = 'lists'
+    members = 'members'
+    labels = 'labels'
+
+
+class TrelloDataSource(ToucanDataSource):
+    board_id: str
+    fields_list: List[Fields] = [Fields.name, Fields.url,
+                                 Fields.lists, Fields.members,  Fields.labels]
+    custom_fields: bool = True
+
+
+class TrelloConnector(ToucanConnector):
+    type = "Trello"
+    data_source_model: TrelloDataSource
+
+    key_id: str = ''
+    token: str = ''
+
+    def get_board(self, path, **customParams):
+        return requests.get(f'https://api.trello.com/1/boards/{path}', params={
+            'key': self.key_id,
+            'token': self.token, **customParams
+        }).json()
+
+    def replace_id_by_value(self, card_with_id, lists=None, labels=None,
+                            members=None, custom_fields=None):
+        """
+        `card_with_id` is a dictionary containing all data of a card,
+        but with unreadlable id instead of value
+        This fonction return `card_with_value` with the same dictionnary
+        as `card_with_id` but with readable value
+
+        `lists`: dictionnary of correspondance between list names and ids
```

Comment:
The name of this parameter is a bit confusing : `lists` sounds like a list of lists, but it is actually a dict, call it `names_ids_mapping` for example and then `members_ids_mapping`